plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/client/PinotGrpcDataFetcher.java

```diff
@@ -193,17 +193,17 @@ public TlsGrpcQueryClientFactory(PinotGrpcServerQueryClientConfig grpcClientConf
         {
             requireNonNull(grpcClientConfig, "grpcClientConfig is null");
             requireNonNull(tlsConfig, "tlsConfig is null");
-            this.config = new GrpcQueryClient.Config(ImmutableMap.<String, Object>builder()
-                    .put(CONFIG_MAX_INBOUND_MESSAGE_BYTES_SIZE, String.valueOf(grpcClientConfig.getMaxInboundMessageSize().toBytes()))
-                    .put(CONFIG_USE_PLAIN_TEXT, String.valueOf(grpcClientConfig.isUsePlainText()))
-                    .put(GRPC_TLS_PREFIX + "." + KEYSTORE_TYPE, tlsConfig.getKeystoreType())
-                    .put(GRPC_TLS_PREFIX + "." + KEYSTORE_PATH, tlsConfig.getKeystorePath())
-                    .put(GRPC_TLS_PREFIX + "." + KEYSTORE_PASSWORD, tlsConfig.getKeystorePassword())
-                    .put(GRPC_TLS_PREFIX + "." + TRUSTSTORE_TYPE, tlsConfig.getTruststoreType())
-                    .put(GRPC_TLS_PREFIX + "." + TRUSTSTORE_PATH, tlsConfig.getTruststorePath())
-                    .put(GRPC_TLS_PREFIX + "." + TRUSTSTORE_PASSWORD, tlsConfig.getTruststorePassword())
-                    .put(GRPC_TLS_PREFIX + "." + SSL_PROVIDER, tlsConfig.getSslProvider())
-                    .buildOrThrow());
+
+            ImmutableMap.Builder<String, Object> tlsConfigBuilder = ImmutableMap.builder();
+            tlsConfigBuilder.put(KEYSTORE_TYPE, tlsConfig.getKeystoreType());
+            tlsConfig.getKeystorePath().ifPresent(path -> tlsConfigBuilder.put(KEYSTORE_PATH, path));
+            tlsConfig.getKeystorePassword().ifPresent(password -> tlsConfigBuilder.put(KEYSTORE_PASSWORD, password));
+            tlsConfigBuilder.put(TRUSTSTORE_TYPE, tlsConfig.getTruststoreType());
+            tlsConfig.getTruststorePath().ifPresent(path -> tlsConfigBuilder.put(TRUSTSTORE_PATH, path));
+            tlsConfig.getTruststorePassword().ifPresent(password -> tlsConfigBuilder.put(TRUSTSTORE_PASSWORD, password));
+            tlsConfigBuilder.put(SSL_PROVIDER, tlsConfig.getSslProvider());
```

Comment:
The config seems to allow specifying a keystore and truststore type but without any paths to the actual keystore and trust-store. i.e. should we assume if any one of the keystore/truststore property is specified then at-least the keystore/truststore type and path must be present? password is always optional.

Reply:
I think we can pass the keystore/truststore type even as Optional variable like we do in Kafka connector. Plus `TlsUtils#extractTlsConfig` would use a default values if the specific config properties are missing. @hashhar WDYT ?